### ISSUE TITLE
Error on migration(autoupdate) when I has two or more model with distinct structure, but each use different schema, using the same table name

### ISSUE THREAD

**Issue by FabricioCoutinho** [strongloop/loopback-connector-postgresql#287]
# Description/Steps to reproduce
on the situation below when I run autoupdate occurs the error **"error: column "client_id" of relation "my_orders" does not exist"**.
This  error occurs because the loopback-connector-postgresql dont check the schema on getTableStatus (migration.js)

I create a repo to share the problem (https://github.com/FabricioCoutinho/loopback-postgres-schema-migration)

**model/order-schema1.json**
```javascript
{
  "name": "Order_Schema1",
  "base": "PersistedModel",
  "public": false,
  "idInjection": true,
  "options": {
    "validateUpsert": true,
    "postgresql": {
      "table": "my_orders",
      "schema": "schema1"
    }
  },
  "properties": {
    "active": {
      "type": "boolean",
      "required": false
    },
    "description": {
      "type": "string",
      "required": false
    },
    "client_id": {
      "type": "number",
      "required": false
    }
  }
}
```
**model/order-schema2.json**

```javascript
{
  "name": "Order_Schema2",
  "base": "PersistedModel",
  "public": false,
  "idInjection": true,
  "options": {
    "validateUpsert": true,
    "postgresql": {
      "table": "my_orders",
      "schema": "schema1"
    }
  },
  "properties": {
    "active": {
      "type": "boolean",
      "required": false
    },
    "description": {
      "type": "string",
      "required": false
    }
  }
}
```

**model-config.json**

```javascript
{
  "_meta": {
    "sources": [
      "loopback/common/models",
      "loopback/server/models",
      "../common/models",
      "./models"
    ],
    "mixins": [
      "loopback/common/mixins",
      "loopback/server/mixins",
      "../common/mixins",
      "./mixins"
    ]
  },
  "Order_Schema1": {
    "dataSource": "DsSchema1"
  },
  "Order_Schema2": {
    "dataSource": "DsSchema2"
  }  
}
```
**datasources.json**

```javascript
{
  "DsSchema1": {
    "host": "192.168.33.10",
    "port": 5432,
    "url": "",
    "database": "loopback",
    "password": "root",
    "name": "dsSchema1",
    "user": "root",
    "connector": "postgresql",
    "debug": true
  },
  "DsSchema2": {
    "host": "192.168.33.10",
    "port": 5432,
    "url": "",
    "database": "loopback",
    "password": "root",
    "name": "dsSchema1",
    "user": "root",
    "connector": "postgresql",
    "debug": true
  }
}
```

# Expected result
The loopback connector check the schema on method getTableStatus on migration.js when check the structure of the model.

# Additional information
I fixed the sql on this method

```javascript
    this.execute('SELECT column_name AS "column", data_type AS "type", ' +
      'is_nullable AS "nullable", character_maximum_length as "length"' // , data_default AS "Default"'
      + ' FROM "information_schema"."columns" WHERE table_name=\'' +
      this.table(model) + '\' and table_schema=\'' + 
      this.schema(model) + '\'', decoratedTableDataCallback);

    this.execute(
      'SELECT t.relname AS "table", i.relname AS "name", ' +
      'am.amname AS "type", ix.indisprimary AS "primary", ' +
      'ix.indisunique AS "unique", ' +
      'ARRAY(SELECT pg_get_indexdef(ix.indexrelid, k + 1, true) ' +
      '  FROM generate_subscripts(ix.indkey, 1) AS k ' +
      '  ORDER BY k ) AS "keys", ' +
      'ARRAY(SELECT ' +
      '  CASE ix.indoption[k] & 1 WHEN 1 THEN \'DESC\' ELSE \'ASC\' END ' +
      '  FROM generate_subscripts(ix.indoption, 1) AS k ' +
      '  ORDER BY k ' +
      ') AS "order" ' +
      'FROM pg_class t, pg_class i, pg_index ix, pg_am am, pg_namespace ns ' +
      'WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND ' +
      'i.relam = am.oid AND ' +
      't.relkind=\'r\' AND t.relname=\'' + this.table(model) + '\'' +
      ' and (ns.oid = t.relnamespace and ns.nspname=\'' +
      this.schema(model) + '\')', decoratedIndexDataCallback);
```

**Comment by ssh24** [comment]
@FabricioCoutinho I am not sure what the issue is you are pointing to.

If you specify the **schema** on the model-definition, you can have two different schemas with the same table name, however, the model name needs to be different because loopback will throw error with the same model name.

Here is a sample reproduction:

**employee.json**
```json
{
  "name": "Employee",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true,
    "postgresql": {
      "table": "employee",
      "schema": "schema1"
    }
  },
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "number"
    }
  },
  "validations": [],
  "relations": {},
  "acls": [],
  "methods": {}
}
```

**employee-1.json**
```json
{
  "name": "Employee1",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true,
    "postgresql": {
      "table": "employee",
      "schema": "schema2"
    }
  },
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "number"
    }
  },
  "validations": [],
  "relations": {},
  "acls": [],
  "methods": {}
}
```

**boot-script.js**
```js
'use strict';
var util = require('util');
var _ = require('lodash');

module.exports = function(app) {
    var db = app.datasources.postgresqlDs;
    var Employee = app.models.Employee;
    var Employee1 = app.models.Employee1;

    db.automigrate(function(err) {
        if (err) throw err;
        console.log('\nAutomigrate completed');
    });
};
```

**database output**
```bash
testdb=# \dn
   List of schemas
   Name   |  Owner
----------+----------
 myschema | root
 public   | postgres
 schema1  | root
 schema2  | root
(4 rows)

testdb=# \dt schema1.*
         List of relations
 Schema  |   Name   | Type  | Owner
---------+----------+-------+-------
 schema1 | employee | table | root
(1 row)

testdb=# \dt schema2.*
         List of relations
 Schema  |   Name   | Type  | Owner
---------+----------+-------+-------
 schema2 | employee | table | root
(1 row)
```

Is this what you were trying to achieve?

**Comment by FabricioCoutinho** [comment]
@ssh24 if the model has different properties and run autoupdate its throw an error.

Change your model **employee-1.json**

```javascript
{
  "name": "Employee1",
  "base": "PersistedModel",
  "idInjection": true,
  "options": {
    "validateUpsert": true,
    "postgresql": {
      "table": "employee",
      "schema": "schema2"
    }
  },
  "properties": {
    "name": {
      "type": "string"
    },
    "age": {
      "type": "number"
    },
   "code": {
      "type": "number"     
   }
  },
  "validations": [],
  "relations": {},
  "acls": [],
  "methods": {}
}
```

and your **boot-script.js** to

````javascript
'use strict';
var util = require('util');
var _ = require('lodash');

module.exports = function(app) {
    var db = app.datasources.postgresqlDs;
    var Employee = app.models.Employee;
    var Employee1 = app.models.Employee1;

    db.autoupdate(function(err) {
        if (err) throw err;
        console.log('\nAutomigrate completed');
    });
};
````

An error like this is show **"error: column "code" of relation "employee" does not exist"**.

**Comment by ssh24** [comment]
@FabricioCoutinho The issue has been fixed on #289 and will be released with the next version of the connector. Thank you for reporting the bug.
